File: app/src/main/java/com/sysu/edu/view/ButtonAdapter.java
package com.sysu.edu.view;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.button.MaterialButtonGroup;
import com.sysu.edu.databinding.ItemButtonGroupBinding;
import com.sysu.edu.databinding.ItemButtonOutlineBinding;

import java.util.ArrayList;

public class ButtonAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    
    private final ArrayList<String> data = new ArrayList<>();
    private onBindListener onBindListener;
    
    public void setListener(onBindListener onBindListener) {
        this.onBindListener = onBindListener;
    }
    
    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        MaterialButtonGroup itemView = ItemButtonGroupBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false).getRoot();
        for (int i = 0; i < data.size(); i++) {
            ItemButtonOutlineBinding binding = ItemButtonOutlineBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
            binding.getRoot().setText(data.get(i));
            if (onBindListener != null) onBindListener.onBind(binding.getRoot(), i);
            itemView.addView(binding.getRoot());
        }
        return new RecyclerView.ViewHolder(itemView) {
        };
    }
    
    public void add(String text) {
        data.add(text);
        notifyItemChanged(0);
    }
    
    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
    }
    
    @Override
    public int getItemCount() {
        return 1;
    }
    
    public interface onBindListener {
        void onBind(Button button, int position);
    }
}

File: app/src/main/java/com/sysu/edu/view/ButtonAdapter.kt
package com.sysu.edu.view

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Button
import androidx.recyclerview.widget.RecyclerView
import com.sysu.edu.databinding.ItemButtonGroupBinding
import com.sysu.edu.databinding.ItemButtonOutlineBinding

class ButtonAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder?>() {
	private val data = mutableListOf<String?>()
	private var onBindListener: OnBindListener? = null
	
	fun setListener(listener: OnBindListener?) {
		onBindListener = listener
	}
	
	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
		val itemView = ItemButtonGroupBinding.inflate(LayoutInflater.from(parent.context), parent, false).root
		data.forEachIndexed { i, d ->
			itemView.addView(ItemButtonOutlineBinding.inflate(LayoutInflater.from(parent.context), parent, false).root.apply {
				text = d
				onBindListener?.onBind(this, i)
			})
		}
		return object : RecyclerView.ViewHolder(itemView) {}
	}
	
	fun add(text: String?) {
		data.add(text)
		notifyItemChanged(0)
	}
	
	override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
	}
	
	override fun getItemCount(): Int = 1
}

fun interface OnBindListener {
	fun onBind(button: Button, position: Int)
}